churaverse-plugins-client/src/churarenPlugin/churarenPlayerPlugin/churarenPlayerPlugin.ts

```diff
@@ -0,0 +1,281 @@
+    this.itemPluginStore.items.delete(ev.itemId)
+    this.itemPluginStore.itemRenderers.delete(ev.itemId)
+    if (ev.playerId === this.playerPluginStore.ownPlayerId) {
+      this.updateMaterialItemBox(ev.playerId)
+    }
+  }
+
+  private readonly playerItemChase = (ev: PlayerWalkEvent): void => {
+    const player = this.playerPluginStore.players.get(ev.id)
+    if (player === undefined) return
+    const speed = ev.speed ?? GRID_SIZE / GRID_WALK_DURATION_MS
+
+    // TODO: 錬金アイテムを持っている場合の処理を追加
+    const items = this.playerItemStore.materialItems.getAllItem(ev.id)
+    items.forEach((item: Item, index: number) => {
+      if (item === undefined) return
+      const renderer = this.playerItemStore.materialItemRenderers.get(item.itemId)
+      const dest = index === 0 ? player.position.copy() : items[index - 1].position.copy()
+      renderer?.chase(dest, speed, (pos: { x: number; y: number }) => {
+        item.position.x = pos.x
+        item.position.y = pos.y
+      })
+    })
+  }
+
+  private readonly dropItem = (ev: DropChurarenItemEvent): void => {
+    const renderer = this.playerItemStore.materialItemRenderers.get(ev.itemId)
+    if (renderer == null) return
+    renderer.destroy()
+    this.playerItemStore.materialItemRenderers.delete(ev.itemId)
+    this.playerItemStore.materialItems.delete(ev.playerId, ev.itemId)
+
+    if (ev.playerId === this.playerPluginStore.ownPlayerId) {
+      this.updateMaterialItemBox(ev.playerId)
+      const dropItemData: DropChurarenItemData = { playerId: ev.playerId, itemId: ev.itemId }
+      const dropItemMessage = new DropChurarenItemMessage(dropItemData)
+      this.networkPluginStore.messageSender.send(dropItemMessage)
+    }
+  }
+
+  private readonly logPlayerDeathByBoss = (ev: LivingDamageEvent): void => {
+    // TODO: プレイヤーがボスによって死亡した時のログを流す
+    //  (churarenBossPluginのChurarenEnemyDamageCauseかどうかを判定する)
+  }
+
+  private readonly onChangePlayerName = (ev: PlayerNameChangeEvent): void => {
+    if (!this.isActive) return
+    this.updateGhostPlayerList()
+  }
+
+  private clearPlayerItemBox(playerId?: string): void {
+    if (playerId === undefined) {
+      this.playerItemStore.materialItemRenderers.forEach((renderer) => {
+        renderer.destroy()
+      })
+      this.playerItemStore.materialItems.clearAll()
+      // TODO: 錬金アイテムの削除処理を追加
+    } else {
+      const materialItems = [...this.playerItemStore.materialItems.getAllItem(playerId)]
+      materialItems.forEach((item) => {
+        const renderer = this.playerItemStore.materialItemRenderers.get(item.itemId)
+        if (renderer !== undefined) {
+          renderer.destroy()
+          this.playerItemStore.materialItemRenderers.delete(item.itemId)
+          this.playerItemStore.materialItems.delete(playerId, item.itemId)
+        }
+      })
+
+      // TODO: 錬金アイテムの削除処理を追加
+    }
+  }
+
+  private updateMaterialItemBox(playerId: string): void {
+    const itemBoxes = this.playerItemStore.materialItems.getAllItem(playerId)
```

Comment:
https://github.com/churadata/churaverse-plugins/pull/39#discussion_r2242013485

こちらも同様にitemsでいい気がします
```suggestion
    const items = this.playerItemStore.materialItems.getAllItem(playerId)
```